test/unit/views/SupportingMaterialFileListViewSpec.scala

```diff
@@ -17,26 +17,73 @@
 package views
 
 import forms.SupportingMaterialFileListFormProvider
-import models.NormalMode
+import models.{FileAttachment, NormalMode}
 import play.api.data.Form
+import play.twirl.api.HtmlFormat
 import views.behaviours.YesNoViewBehaviours
 import views.html.supportingMaterialFileList
 
 class SupportingMaterialFileListViewSpec extends YesNoViewBehaviours {
 
-  val messageKeyPrefix = "supportingMaterialFileList"
+  private lazy val messageKeyPrefix = "supportingMaterialFileList"
 
-  val form = new SupportingMaterialFileListFormProvider()()
+  override protected val form = new SupportingMaterialFileListFormProvider()()
 
-  def createView = () => supportingMaterialFileList(frontendAppConfig, form, Seq.empty, NormalMode)(fakeRequest, messages)
+  private def createView: () => HtmlFormat.Appendable = { () =>
+    createViewWithForm(form)
+  }
 
-  def createViewUsingForm = (form: Form[_]) => supportingMaterialFileList(frontendAppConfig, form, Seq.empty, NormalMode)(fakeRequest, messages)
+  private def createViewWithForm(f: Form[Boolean], files: Seq[FileAttachment] = Seq.empty): HtmlFormat.Appendable = {
+    supportingMaterialFileList(frontendAppConfig, f, files, NormalMode)(fakeRequest, messages)
+  }
 
   "SupportingMaterialFileList view" must {
 
     behave like normalPage(createView, messageKeyPrefix)()
 
     behave like pageWithBackLink(createView)
 
+//    behave like yesNoPage() // TODO: not working at the moment as it has not been created as the other Yes/No pages
```

Comment:
@pllaurado @SteveSugden 
I feel like the refactoring has not done properly for this page.
I am waiting for your opinion here.

Reply:
I think Pol's opinion is more valid than mine.   I suspect the reason is that this is not a simple "Yes/No goto next page" page.

Reply:
It is a Yes/No behaviour.
The endpoint is: `/supportingMaterialFileList`

it is not a simple page, but we did not defined it properly as the other boolean questions...

Also look here: https://github.com/hmrc/binding-tariff-trader-frontend/blob/master/test/unit/views/SupportingInformationViewSpec.scala#L26

It says: `SupportingInformationViewSpec extends YesNoViewBehaviours`